unity-client/Assets/Scripts/MainScripts/DCL/TheGraph/TheGraph.cs

```diff
@@ -0,0 +1,129 @@
+using UnityEngine.Networking;
+
+public enum TheGraphCache
+{
+    DontUseCache,
+    UseCache
+}
+
+public interface ITheGraph
+{
+    Promise<string> Query(string url, string query);
+    Promise<string> Query(string url, string query, QueryVariablesBase variables);
+    Promise<List<Land>> QueryLands(string tld, string address, TheGraphCache cache);
+}
+
+public class TheGraph : ITheGraph
+{
+    private const float CACHE_TIME = 5 * 60;
+
+    private readonly Dictionary<string, QueryLandCache> landQueryCache = new Dictionary<string, QueryLandCache>();
+
+    public Promise<string> Query(string url, string query) { return Query(url, query, null); }
+
+    public Promise<string> Query(string url, string query, QueryVariablesBase variables)
+    {
+        Promise<string> promise = new Promise<string>();
+
+        if (string.IsNullOrEmpty(query) || string.IsNullOrEmpty(url))
+        {
+            promise.Reject($"error: {(string.IsNullOrEmpty(url) ? "url" : "query")} is empty");
+            return promise;
+        }
+
+        string queryJson = $"\"query\":\"{Regex.Replace(query, @"\p{C}+", string.Empty)}\"";
+        string variablesJson = variables != null ? $",\"variables\":{JsonUtility.ToJson(variables)}" : string.Empty;
+        string bodyString = $"{{{queryJson}{variablesJson}}}";
+
+        var request = new UnityWebRequest();
+        request.url = url;
+        request.method = UnityWebRequest.kHttpVerbPOST;
+        request.downloadHandler = new DownloadHandlerBuffer();
+        request.uploadHandler = new UploadHandlerRaw(string.IsNullOrEmpty(bodyString) ? null : Encoding.UTF8.GetBytes(bodyString));
+        request.SetRequestHeader("Content-Type", "application/json");
+        request.timeout = 60;
+
+        var operation = request.SendWebRequest();
+
+        operation.completed += asyncOperation =>
+        {
+            if (request.WebRequestSucceded())
+            {
+                promise.Resolve(request.downloadHandler.text);
+            }
+            else
+            {
+                promise.Reject($"error: {request.error} response: {request.downloadHandler.text}");
+            }
+            request.Dispose();
+        };
+
+        return promise;
+    }
+
+    public Promise<List<Land>> QueryLands(string tld, string address, TheGraphCache cache = TheGraphCache.UseCache)
+    {
+        string lowerCaseAddress = address.ToLower();
+        
+        Promise<List<Land>> promise = new Promise<List<Land>>();
+
+        if (cache == TheGraphCache.UseCache)
+        {
+            if (landQueryCache.TryGetValue(lowerCaseAddress, out QueryLandCache cacheValue))
+            {
+                if (Time.unscaledTime - cacheValue.lastUpdate <= CACHE_TIME)
+                {
+                    promise.Resolve(cacheValue.lands);
+                    return promise;
+                }
+            }
+        }
+
+        string url = "https://api.thegraph.com/subgraphs/name/decentraland/land-manager";
```

Comment:
Maybe it would be clearer to store this url in a constant.

Reply:
done